src/charts/helpers/component-registry.js

```diff
@@ -0,0 +1,47 @@
+import {rebind} from "./common"
+
+export default function ComponentRegistry () {
+  let components = {}
+  const events = {}
+
+  function capitalizeFirstLetter(string) {
+    return string[0].toUpperCase() + string.slice(1);
+  }
+
+  function register (_components) {
+    components = _components
+    for (let index in components) {
```

Comment:
I wouldn't call this variable `index` here as it's actually a `key`. IMO `index` implies an order, and keys in Objects are not guaranteed to be in a certain order. If you want to have insertion order preserved you could use an [ES6 Map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map), or D3's version of Map. 

With an ES6 Map you can do `myMap.forEach((key, value) => {...})` or `for (let [key, value] of myMap) { ... }`

Reply:
good point